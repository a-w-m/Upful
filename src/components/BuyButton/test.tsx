import React from "react"
import { render, screen } from "@testing-library/react"
import BuyButton from "."

const props = {
  "data-item-id": "",
  "data-item-price": "10",
  "data-item-name": "",
  "data-item-description": "",
  "data-item-image": "",
  "data-item-url": "",
  "data-item-custom1-name": "",
  "data-item-custom1-options": "",
  "data-item-custom1-value": "",
  "data-item-max-quantity": 1,
  productOptions: [],
  optionsSelected: null,
}

describe("BuyButton Component", () => {

  it("matches snapshot", () => {
    render(<BuyButton {...props}></BuyButton>)
    const button = screen.getByRole("button")
    expect(button).toMatchSnapshot()
  })

  it("should not be disabled if max-quantity is greater than 0", () => {
    render(<BuyButton {...props}></BuyButton>)
    const button = screen.queryByRole("button")
    expect(button).not.toBeDisabled()
  })

  it("should display 'Add to Cart' if max-quantity is greater than 0", () => {
    render(<BuyButton {...props}></BuyButton>)
    const button = screen.getByRole("button", {name: /Add to Cart/})
    expect(button).toBeInTheDocument()
  })

  it("should not display 'Sold Out' if max-quantity is greater than 0", () => {
    render(<BuyButton {...props}></BuyButton>)
    const button = screen.queryByRole("button", {name: /Sold Out/})
    expect(button).toBeNull()
  })

  it("should be disabled if max-quantity is not greater than 0", () => {
    render(
      <BuyButton {...props} data-item-max-quantity={0}></BuyButton>
    )
    const button = screen.getByRole("button")

    expect(button).toBeDisabled()
  })

  it("component renders 'Sold Out' if max-quantity is not greater than 0", () => {
    render(
      <BuyButton {...props} data-item-max-quantity={0}></BuyButton>
    )
    const button = screen.getByRole("button", {name: /Sold Out/})

    expect(button).toBeInTheDocument()
  })

  it("component does not render 'Add to Cart' if max-quantity is not greater than 0", () => {
   render(
      <BuyButton {...props} data-item-max-quantity={0}></BuyButton>
    )
    const button = screen.queryByRole("button", {name: /Add to Cart/})

    expect(button).toBeNull()
  })
})
