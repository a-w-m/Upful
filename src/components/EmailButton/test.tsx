import React from "react"
import { render, screen } from "@testing-library/react"
import EmailButton from "."

const props = {
  title: "test-email",
}

describe("Email", () => {
  beforeEach(() => {
    render(<EmailButton {...props}></EmailButton>)
  })

  it("should match snapshot", () => {
    const link = screen.getByRole("link", { name: /Email/i })
    expect(link).toMatchSnapshot()
  })

  it("should display icon", ()=>{
    const icon = screen.getByRole("img")
    expect(icon).toBeInTheDocument()
  })

  it("should display descriptive text", () => {
    const description = screen.getByText(/Email us about this product/)
    expect(description).toBeInTheDocument()
  })

  it("should have correct href", () => {
    const link = screen.getByRole("link")

    expect(link).toHaveAttribute(
      "href",
      `mailto:howdy@upful.com?Subject=${props.title}`
    )
  })
})
