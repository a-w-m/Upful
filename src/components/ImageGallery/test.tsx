import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import ImageGallery from "./index"
import { imageGalleryData } from "../../../__mocks__/mock-data"

describe("ImageGallery Component", () => {
  const mockDispatch = jest.fn()

  test("matches snapshot", () => {
    const { container } = render(
      <ImageGallery
        images={imageGalleryData}
        dispatch={mockDispatch}
      ></ImageGallery>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test("dispatch is called when gallery image is clicked ", () => {
    render(
      <ImageGallery
        images={imageGalleryData}
        dispatch={mockDispatch}
      ></ImageGallery>
    )
    const image = screen.getAllByRole("listitem")[0]
    fireEvent.click(image)
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})

test("no list rendered when no images supplied", () => {
  const { container } = render(
    <ImageGallery images={[]} dispatch={jest.fn()}></ImageGallery>
  )
  expect(container.firstChild).toBeNull()
})
