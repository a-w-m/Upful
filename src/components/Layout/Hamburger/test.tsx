import React from "react"
import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Hamburger from "."

describe("Hamburger component", () => {
  const setOpen = jest.fn()

  it("should match snapshot", () => {
    render(<Hamburger open={false} setOpen={setOpen} />)
    const button = screen.getByRole("button")
    expect(button).toMatchSnapshot()
  })

  it("should handle click", async () => {
    render(<Hamburger open={false} setOpen={setOpen} />)
    const user = userEvent.setup()
    const button = screen.getByRole("button")
    await user.click(button)
    expect(setOpen).toHaveBeenCalledTimes(1)
  })

  it("should apply style when open", () => {
    render(<Hamburger open={true} setOpen={setOpen} />)
    const menu = document.querySelectorAll("span")
    expect(menu[0]).toHaveStyle({ transform: "rotate(45deg)" })
    expect(menu[1]).toHaveStyle({ opacity: "0", transform: "translateX(2rem)" })
    expect(menu[2]).toHaveStyle({ transform: "rotate(-45deg)" })
  })

  it("should apply style when not open", () => {
    render(<Hamburger open={false} setOpen={setOpen} />)
    const menu = document.querySelectorAll("span")
    expect(menu[0]).toHaveStyle({ transform: "rotate(0)" })
    expect(menu[1]).toHaveStyle({ opacity: "1", transform: "translateX(0)" })
    expect(menu[2]).toHaveStyle({ transform: "rotate(0)" })
  })
})
