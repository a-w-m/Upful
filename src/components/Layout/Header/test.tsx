import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import Header from ".."

describe("Header component", () => {
  test("matches snapshot", () => {
    const { container } = render(<Header children="" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test("renders Header with correct site title", () => {
    render(<Header children="" />)

    expect(screen.getByText("StoreFront")).toBeInTheDocument()
  })

  // test("displays toggle menu on small screens", ()=>{
  //   render(<Header children="" />)

  //   expect(screen.getByAltText("toggle-menu")).toBeInTheDocument()

  // })

  // test("does not display toggle menu on large screens", ()=>{
  //   render(<Header children="" />)

  //   expect(screen.queryAllByAltText("toggle-menu")).toBeNull()

  // })
})

// describe("click outside toggle menu", () => {
//   test("Nav remains visible when clicking inside component ", () => {
//     render(<Header children="" />)
//     const Button = screen.getByRole("button")
//     const Nav = screen.getAllByRole("navigation", { hidden: true })[0]
//     fireEvent.click(Button)
//     fireEvent.click(Nav)
//     expect(Nav).toBeVisible()
//   })

//   test("Nav is not visible when clicking outside component", () => {
//     render(<Header children="" />)
//     const Button = screen.getByRole("button")
//     const Nav = screen.getAllByRole("navigation", { hidden: true })[0]
//     fireEvent.click(Button)
//     fireEvent.mouseDown(Button)
//     expect(Nav).not.toBeVisible()
//   })
// })
