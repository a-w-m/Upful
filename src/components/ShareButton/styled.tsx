import styled from 'styled-components'
import device from '../layout/mediaQuery'


export const ShareWrapper = styled.div`
position:relative;
`

export const Button = styled.button<{open?: boolean}>`
    display: flex;
    justify-content: center;
    gap: 1.6rem;
    align-items: end;
    background: none;
    border: none;
    cursor: pointer;
    padding-top: 1.6rem;
    width: 100%;



    & > a{
        cursor: pointer;
        color: inherit;
        text-decoration: none;
    }
`

export const TextWrapper = styled.span`
    font-size: 1.4rem;

    @media ${device.laptop}{
    font-size: 1.4rem;
    }

`

export const SocialsContainer = styled.div<{open: boolean}>`
    display: flex;
    visibility: ${({open})=> open? "visible" : "hidden"};
    opacity: ${({open})=> open? "1" : "0"};
    pointer-events: ${({open})=> open? "auto" : "none"};
    justify-content: space-evenly;
    align-items: center;
    position: absolute;
    bottom: 0;
    left: 0%;
    width: 100%;
    height: 50%;
    z-index: 11;
    box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.08);
    background-color: #fff;
    transition: 200ms linear; 

    & > a {
        color: inherit;
    }

    & > a:hover{
        color: var(--main-brown)
    }

    @media ${device.laptop}{
    }
    
`