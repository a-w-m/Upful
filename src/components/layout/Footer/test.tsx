import React from 'react'
import {render, screen} from '@testing-library/react'
import Footer from "."


describe("Footer Component", ()=>{

    test("matches component", ()=>{
        const {container} = render(<Footer />)
        expect(container.firstElementChild).toMatchSnapshot()

    })

})