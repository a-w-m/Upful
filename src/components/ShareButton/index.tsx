import React, { useState, useRef } from "react"
import { ShareIcon } from "../Icons/index"
import { Button, TextWrapper, SocialsContainer } from "./styled"
import { Facebook, Twitter, Pinterest } from "../Icons/index"
import useOnClickOutside from "../../hooks/useOnClickOutside"

interface ShareButtonProps {
  title: string
  path: string
  imageURL: string
}

const ShareButton: React.FC<ShareButtonProps> = props => {
  const { title, path, imageURL } = props
  const ref = useRef(null)
  const [isOpen, setOpen] = useState(false)
  useOnClickOutside(ref, () => setOpen(false))

  return (
    <>
      <Button
        open={isOpen}
        type="button"
        onClick={() => {
          setOpen(true)
        }}
      >
        <ShareIcon />
        <TextWrapper>Share This</TextWrapper>
        <SocialsContainer ref={ref} open={isOpen}>
          <a
            href={`http://www.facebook.com/sharer.php?u=${path}`}
            target="_blank"
          >
            <Facebook />
          </a>
          <a
            href={`https://twitter.com/intent/tweet?url=${path}`}
            target="_blank"
          >
            <Twitter />
          </a>
          <a
            href={`http://pinterest.com/pin/create/button/?url=${path}&media=${imageURL}&description=${title}`}
            target="_blank"
          >
            <Pinterest />
          </a>
        </SocialsContainer>
      </Button>
    </>
  )
}

export default ShareButton
