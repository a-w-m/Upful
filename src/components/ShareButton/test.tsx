import React from "react"
import { render, screen, waitFor, within } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import ShareButton from "."
import { getQueryMockData } from "src/utils/test/data"

const { siteMetadata } = getQueryMockData.getSiteQuery().site

beforeEach(() => {
  render(
    <ShareButton
      title={siteMetadata.title}
      path={siteMetadata.url}
      imageURL={siteMetadata.image}
    />
  )
})

describe("Before user clicks ShareButton", () => {
  it("should match snapshot", () => {
    const button = screen.getByRole("button", { name: /Share Product/ })
    expect(button).toMatchSnapshot()
  })
  it("should display button", () => {
    const button = screen.getByRole("button", { name: /Share Product/ })
    expect(button).toBeVisible()
  })

  it("should display share icon", () => {
    expect(screen.getByTestId(/Share Icon/)).toBeVisible()
  })

  it("should not display Facebook link", () => {
    const facebook = screen.queryByRole("link", { name: /Share to Facebook/ })
    expect(facebook).toBeNull()
  })

  it("should not display Twitter link", () => {
    const twitter = screen.queryByRole("link", { name: /Share to Twitter/ })
    expect(twitter).toBeNull()
  })

  it("should not display Pinterest link", () => {
    const pinterest = screen.queryByRole("link", { name: /Share to Pinterest/ })
    expect(pinterest).toBeNull()
  })
})

describe("After user clicks ShareButton", () => {
  it("should display Facebook link", async () => {
    const user = userEvent.setup()
    const button = screen.getByRole("button", { name: /Share Product/ })
    await user.click(button)
    const link = screen.getByRole("link", { name: /Share to Facebook/ })
    expect(link).toBeVisible()
  })

  it("should have correct Facebook url", async () => {
    const user = userEvent.setup()
    const button = screen.getByRole("button", { name: /Share Product/ })
    await user.click(button)
    const link = screen.getByRole("link", { name: /Share to Facebook/ })
    expect(link).toHaveAttribute(
      "href",
      `http://www.facebook.com/sharer.php?u=${siteMetadata.url}`
    )
  })

  it("should display Facebook icon", async () => {
    const user = userEvent.setup()
    const button = screen.getByRole("button", { name: /Share Product/ })
    await user.click(button)
    const link = screen.getByRole("link", { name: /Share to Facebook/ })
    expect(within(link).getByTestId(/Facebook Icon/)).toBeVisible()
  })

  it("should display Twitter link", async () => {
    const user = userEvent.setup()
    const button = screen.getByRole("button", { name: /Share Product/ })
    await user.click(button)
    const link = screen.getByRole("link", { name: /Share to Twitter/ })
    expect(link).toBeVisible()
  })

  it("should have correct Twitter url", async () => {
    const user = userEvent.setup()
    const button = screen.getByRole("button", { name: /Share Product/ })
    await user.click(button)
    const link = screen.getByRole("link", { name: /Share to Twitter/ })
    expect(link).toHaveAttribute(
      "href",
      `https://twitter.com/intent/tweet?url=${siteMetadata.url}`
    )
  })

  it("should display Twitter icon", async () => {
    const user = userEvent.setup()
    const button = screen.getByRole("button", { name: /Share Product/ })
    await user.click(button)
    const link = screen.getByRole("link", { name: /Share to Twitter/ })
    expect(within(link).getByTestId(/Twitter Icon/)).toBeVisible()
  })

  it("should display Pinterest link", async () => {
    const user = userEvent.setup()
    const button = screen.getByRole("button", { name: /Share Product/ })
    await user.click(button)
    const link = screen.getByRole("link", { name: /Share to Pinterest/ })
    expect(link).toBeVisible()
  })

  it("should have correct Pinterest url", async () => {
    const user = userEvent.setup()
    const button = screen.getByRole("button", { name: /Share Product/ })
    await user.click(button)
    const link = screen.getByRole("link", { name: /Share to Pinterest/ })
    expect(link).toHaveAttribute(
      "href",
      `http://pinterest.com/pin/create/button/?url=${siteMetadata.url}&media=${siteMetadata.image}&description=${siteMetadata.title}`
    )
  })

  it("should display Pinterest Icon", async () => {
    const user = userEvent.setup()
    const button = screen.getByRole("button", { name: /Share Product/ })
    await user.click(button)
    const link = screen.getByRole("link", { name: /Share to Pinterest/ })
    expect(within(link).getByTestId("Pinterest Icon")).toBeVisible()
  })

  it("should hide links container when user clicks an outside node", async () => {
    const user = userEvent.setup()
    const button = screen.getByRole("button", { name: /Share Product/ })
    await user.click(button)
    const link = screen.getByRole("link", { name: /Share to Facebook/ })
    expect(link).toBeVisible()
    await user.click(document.body)
    expect(link).not.toBeVisible()
  })

  it("should not hide links container when user clicks within component", async () => {
    const user = userEvent.setup()
    const button = screen.getByRole("button", { name: /Share Product/ })
    await user.click(button)
    const link = screen.getByRole("link", { name: /Share to Facebook/ })
    await user.click(link)
    expect(link).toBeVisible()
  })
})
