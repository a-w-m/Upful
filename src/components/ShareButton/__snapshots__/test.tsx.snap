// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Before user clicks ShareButton should match snapshot 1`] = `
<button
  class="sc-gsnTZi edmMsC"
  type="button"
>
  <svg
    aria-hidden="true"
    data-testid="Share Icon"
    fill="none"
    focusable="false"
    height="21"
    viewBox="0 0 18 21"
    width="27"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Share Product
    </title>
    <path
      clip-rule="evenodd"
      d="M9.545.56L9 0l-.545.56-5.143 5.286 1.09 1.122L8.23 3.035V15.45H9.77V3.035l3.826 3.933 1.091-1.122L9.545.561zM0 11.75V21h18v-9.25h-1.543v7.664H1.543V11.75H0z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
  <span
    class="sc-dkzDqf eWJTSj"
  >
    Share Product
  </span>
  <div
    class="sc-hKMtZM fklTcG"
  >
    <a
      aria-label="Share to Facebook"
      href="http://www.facebook.com/sharer.php?u=upful.shop"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="svg-icon icon-share-facebook "
        data-testid="Facebook Icon"
        fill="none"
        focusable="false"
        height="18"
        viewBox="0 0 9 18"
        width="9"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Share to Facebook
        </title>
        <path
          clip-rule="evenodd"
          d="M7.861 9.007H5.394V18H1.737V9.007H0V5.829h1.738V3.772C1.737 2.3 2.421 0 5.427 0l2.709.01v3.085H6.17c-.321 0-.775.165-.775.866v1.868h2.786l-.32 3.178z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </a>
    <a
      aria-label="Share to Twitter"
      href="https://twitter.com/intent/tweet?url=upful.shop"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="svg-icon icon-share-twitter "
        data-testid="Twitter Icon"
        fill="none"
        focusable="false"
        height="20"
        viewBox="0 0 20 17"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Share to Twitter
        </title>
        <path
          clip-rule="evenodd"
          d="M17.64 2.576a4.115 4.115 0 0 0 1.803-2.27c-.792.47-1.67.812-2.605.996a4.102 4.102 0 0 0-6.99 3.742A11.647 11.647 0 0 1 1.392.757 4.083 4.083 0 0 0 .837 2.82a4.1 4.1 0 0 0 1.825 3.413A4.086 4.086 0 0 1 .803 5.72v.052a4.105 4.105 0 0 0 3.29 4.022 4.108 4.108 0 0 1-1.852.07 4.106 4.106 0 0 0 3.832 2.85 8.231 8.231 0 0 1-5.094 1.755A8.33 8.33 0 0 1 0 14.412a11.612 11.612 0 0 0 6.289 1.843c7.545 0 11.671-6.25 11.671-11.672 0-.178-.004-.355-.011-.53a8.339 8.339 0 0 0 2.047-2.124 8.202 8.202 0 0 1-2.357.646h.001z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </a>
    <a
      aria-label="Share to Pinterest"
      href="http://pinterest.com/pin/create/button/?url=upful.shop&media=image&description=Upful"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="svg-icon icon-share-pinterest "
        data-testid="Pinterest Icon"
        fill="none"
        focusable="false"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Share to Pinterest
        </title>
        <path
          clip-rule="evenodd"
          d="M10 0C4.479 0 0 4.477 0 10a9.999 9.999 0 0 0 6.355 9.311c-.087-.79-.166-2.005.035-2.868.183-.781 1.173-4.971 1.173-4.971s-.3-.6-.3-1.485c0-1.388.808-2.429 1.81-2.429.854 0 1.265.641 1.265 1.409 0 .859-.545 2.141-.827 3.33-.238.996.5 1.806 1.48 1.806 1.776 0 3.144-1.873 3.144-4.577 0-2.394-1.72-4.068-4.178-4.068-2.845 0-4.513 2.134-4.513 4.34 0 .859.329 1.78.741 2.281.083.1.094.188.072.288-.075.315-.245.995-.28 1.133-.043.183-.143.223-.334.134-1.248-.581-2.03-2.407-2.03-3.875 0-3.154 2.292-6.05 6.609-6.05 3.468 0 6.165 2.472 6.165 5.775 0 3.446-2.175 6.221-5.191 6.221-1.013 0-1.965-.527-2.292-1.15l-.625 2.378c-.225.868-.834 1.957-1.242 2.62A10 10 0 0 0 9.999 20C15.521 20 20 15.523 20 10 20 4.478 15.521.001 10 .001V0z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </a>
  </div>
</button>
`;
